### PR TITLE
[BugFix#18] Crop overlay wasn't showing on FF

### PR DESCRIPTION
**Pull request by DPr00f**
@DominicTobias can you take a look at this please?
I've created a SVG and that's the used element to create the Clip-Path as Firefox hates the idea of polygon or inset in the clip-path property.
Now the real question is, if we're going with this solution, shouldn't it be the same for WebKit?
Should we remove the `-webkit-clip-path: polygon(...)` and keep it as `-webkip-clip-path: url("#...")`?
Also, I've tested this solution on Safari and having the values as `100%` still renders perfectly.

TL;DR: Give me thoughts, not sure if this solution should be final.

### CODE REVIEW

File: lib/ReactCrop.jsx [hunk 1]
```diff
@@ -509,37 +509,54 @@ var ReactCrop = React.createClass({
 		);
 	},
 
+	arrayDividedBy100(arr, delimeter) {
```
Comment (DominicTobias) [inline lib/ReactCrop.jsx]: We should be able to use es6 default arg for delimeter (and one below)

File: lib/ReactCrop.jsx [hunk 2]
```diff
@@ -509,37 +509,54 @@ var ReactCrop = React.createClass({
...
+				left: pBottomLeft,
+				right: pBottomRight
+			}
+		};
+	},
 
-		if (crop.y + bottom === 100) {
-			bottom -= 0.00001;
-		}
+	getImageClipStyle() {
```
Comment (DominicTobias) [inline lib/ReactCrop.jsx]: It's hard to see on my phone - did you keep this just for comparison? (Is this method unused now?)

File: lib/ReactCrop.jsx [hunk 3]
```diff
@@ -577,6 +594,24 @@ var ReactCrop = React.createClass({
 		}
 	},
 
+	getClipPathHtml() {
+		let polygon = this.getPolygonValues(true);
+		// We used dangerouslySetInnerHTML because react refuses to add the attribute 'clipPathUnits' to the rendered DOM
+		return {
+			__html: '<clipPath id="ReactCropperClipPolygon" clipPathUnits="objectBoundingBox">' +
```
Comment (DominicTobias) [inline lib/ReactCrop.jsx]: Should be able to make this one es6 string and then not need the pluses?

File: lib/ReactCrop.jsx (hunk 1 above)
Comment (DPr00f) [inline lib/ReactCrop.jsx]: Agreed

**Comment by DominicTobias** [comment]
Great job on this, nice thinking on an alternative. That's good it works on safari too without that clip bug. It looks like a good solution!

File: lib/ReactCrop.jsx (hunk 2 above)
Comment (DPr00f) [inline lib/ReactCrop.jsx]: It's still used. But I wonder if we need it at all. We might get away with url#ReactCropClipPath for both webkit and non prefixed version. What do you think?

File: lib/ReactCrop.jsx (hunk 3 above)
Comment (DPr00f) [inline lib/ReactCrop.jsx]: It would be a really long line. Kept it as is for readability

Comment (DominicTobias) [inline lib/ReactCrop.jsx]: Sure, but you can have line breaks with es6 strings with no need for + or \ :)

File: lib/ReactCrop.jsx (hunk 2 above)
Comment (DominicTobias) [inline lib/ReactCrop.jsx]: Ah yeah I see, yeah if it works on iOS Safari that would be great. If you don't have an iOS phone I'll check it when I'm back

File: lib/ReactCrop.jsx (hunk 3 above)
Comment (DPr00f) [inline lib/ReactCrop.jsx]: Did not know that. Will change it then. Thanks for the tip

File: lib/ReactCrop.jsx (hunk 2 above)
Comment (DPr00f) [inline lib/ReactCrop.jsx]: Yeah, didn't check it. Will do tomorrow.

Comment (DPr00f) [inline lib/ReactCrop.jsx]: Actually, just check it, both Chrome and iOS do not update the clipping path if just the id of the path is set. And I really don't want to force a repaint to see the changes. So will keep it, but will try to clean it up a bit

**Comment by DPr00f** [comment]
Cleaned up. Double check when you can @DominicTobias .
Unfortunately we really can't get rid of the polygon, otherwise safari and chrome don't repaint the view.

**Comment by DominicTobias** [comment]
Nice one, looks good to me

**Comment by DPr00f** [comment]
Cool. Will merge it then

**Comment by DominicTobias** [comment]
I also just thought of a potential alternative, I'm just writing it here so I don't forget as it might be simpler to implement and have good browser compatibility: border-width allows a different width for all sides, so it could be a biggass border

**Comment by DPr00f** [comment]
It's worth to try, just don't like the idea of working out the actual pixels size of the border, I do prefer the idea of having it as percentages. But let's try it.

**Comment by DominicTobias** [comment]
Ah, maybe that's why I didn't do it - yeah if it's not possible to use
percentages then it doesn't sound viable
